types/react-redux/index.d.ts

```diff
@@ -19,6 +19,10 @@ interface ComponentDecorator<TStateProps, TDispatchProps, TOwnProps> {
     (component: Component<TStateProps & TDispatchProps & TOwnProps>): ComponentClass<TOwnProps>;
 }
 
+interface MergedComponentDecorator<TOwnProps, TMergedProps> {
```

Comment:
Could you combine this and the above decorator into a single type, perhaps just `ComponentDecorator`? I think it'd be simpler, and you wouldn't have to juggle `void` anymore.